Repo.rb

```diff
@@ -42,17 +42,18 @@ def self.init_from_git(dir)
     )
     repo_info.write_to_disk!
   end
+
   # Clone a rgit repository and it's subrepos recursively into the passed
   # directory, with the passed options as a string to git
   # Return an object representing the Rgit repository
-  def self.clone(repository, directory, git_str) 
-    system("git clone #{git_str}")
+  def self.clone(repository, directory, opt_str)
+    git_command('clone', "#{opt_str} #{repository}", directory)
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.